### PR TITLE
#48 Fix Audit Issues

### PR DESCRIPTION
**Pull request by mich-master**

### CODE REVIEW

File: maintenance/program/src/state.rs [hunk 1]
```diff
@@ -19,6 +19,7 @@ pub enum MaintenanceAccountType {
 #[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
 pub struct MaintenanceRecord {
     pub account_type: MaintenanceAccountType,
+    pub address: Pubkey,
```
Comment (s-medvedev) [inline maintenance/program/src/state.rs:22]: It is unclear what address is saved in this field.

File: maintenance/program/src/processor.rs [hunk 2]
```diff
@@ -301,27 +310,35 @@ pub fn process_close_maintenance(
 ) -> ProgramResult {
     let account_info_iter = &mut accounts.iter();
 
-    let maintenance_record_info = next_account_info(account_info_iter)?; // 0
-    let program_data_info = next_account_info(account_info_iter)?; // 1
-    let authority_info = next_account_info(account_info_iter)?; // 2
-    let spill_info = next_account_info(account_info_iter)?; // 3
+    let bpf_loader_program_info = next_account_info(account_info_iter)?; // 0
```
Comment (s-medvedev) [inline maintenance/program/src/processor.rs:313]: Let's check that `bpf_loader_program_info.key` equals the BPF loader upgradable program.

File: maintenance/program/src/processor.rs [hunk 3]
```diff
@@ -301,27 +310,35 @@ pub fn process_close_maintenance(
...
+    let maintenance_record_info = next_account_info(account_info_iter)?; // 1
+    let maintained_program_info = next_account_info(account_info_iter)?; // 2
```
Comment (s-medvedev) [inline maintenance/program/src/processor.rs:315]: Let's check that the user specifies the correct `mainteined_program_info.key`.

File: maintenance/program/src/instruction.rs [hunk 4]
```diff
@@ -240,7 +240,9 @@ pub fn close_maintenance(
     let (maintenance_record, _): (Pubkey, u8) = get_maintenance_record_address(program_id, program_address);
 
     let accounts = vec![
+        AccountMeta::new_readonly(bpf_loader_upgradeable::id(), false),
```
[the remark this answers is not in the document]
Comment (mich-master) [reply maintenance/program/src/instruction.rs:243]: Fixed

File: maintenance/program/src/processor.rs (hunk 2 above)
Comment (mich-master) [reply maintenance/program/src/processor.rs:313]: Fixed

File: maintenance/program/src/processor.rs (hunk 3 above)
Comment (mich-master) [reply maintenance/program/src/processor.rs:316]: Already Checked

**Review by s-medvedev — APPROVED** [review]

**Pull request merged**